software/lamp-os/src/core/compositor.cpp

```diff
@@ -7,14 +7,34 @@ namespace lamp {
 Compositor::Compositor() {};
 
 void Compositor::begin(std::vector<AnimatedBehavior*> inBehaviors, std::vector<FrameBuffer*> inFrameBuffers, bool homeMode) {
+  // Delete all behaviors we previously allocated to prevent memory leak
+  for (auto* behavior : ownedBehaviors) {
```

Comment:
This makes so little sense. Why are we deleting things in begin? What memory leak? Something seems weird about how stuff is initializing